src/script/engine.zig

```diff
@@ -312,6 +314,16 @@ pub const Engine = struct {
         try self.stack.pushByteArray(first);
     }
 
+    /// OP_2ROT: The fifth and sixth items back are moved to the top of the stack
+    fn op2Rot(self: *Engine) !void {
+        const index = self.stack.items.items.len - 6;
+        var i: usize = 2;
+        while (i > 0) : (i -= 1) {
+            const value = self.stack.items.orderedRemove(index);
+            try self.stack.pushElement(value);
+        }
```

Comment:
the swap impl you did before was way better in term of performance. I avoid shifting the array

Reply:
you can do it with swap while targeting the end of the slice